teamcity/project_feature_connection.go

```diff
@@ -0,0 +1,161 @@
+package teamcity
```

Comment:
May I suggest we break the `feature_connection` from the specific `Vault` types here?
Perhaps a `project_connection_vault_oauth.go` would be more suitable.